WebDevStudios/wd_s
# Feature/#519 frontend dependencies

Closes #519

### DESCRIPTION ###
Adds [frontend-dependencies](https://www.npmjs.com/package/frontend-dependencies) to manage our... _frontend dependencies_.

This gets rid of the manual copy/paste script we had running and smartly copies our frontend deps over into an `/assets/` directory (this name can be anything we want, if we want to change it).

I haven't been able to figure out if there's a way to install packages to the `frontendDependencies` object the same way you install them to `devDependencies`/`dependencies`. Maybe I'm just missing something, but I wanted to at least get a PR spun up that we can work from.

Right now, if you add your desired package to the object it'll be installed when you run `npm i` or `npm run build`.

This also works well with our `clean` script, so if we update the version number of a frontend dependency we'll just need to run `npm i` or `npm run build` to clear out the entirety of `/assets/` and install the new versions.

### SCREENSHOTS ###
![package file](https://d.pr/free/i/4PoE7g+)

![directory structure](https://d.pr/free/i/65xJWN+)

### OTHER ###

### STEPS TO VERIFY ###
1. Checkout the branch
2. Run `npm i`
3. Rejoice
4. Run `npm run build` to see the clean and install processes run
5. Test adding some other packages to the `frontendDependencies` object to see multiple packages copied over.

### DOCUMENTATION ###
Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?

For sure. We'll need to detail how to install packages with this.
